tables/AlternateMans/test/tColumnarFile.cc

```diff
@@ -0,0 +1,164 @@
+#include <boost/test/unit_test.hpp>
+
+#include "../StokesIStMan/SimpleColumnarFile.h"
+#include "../StokesIStMan/BufferedColumnarFile.h"
+
+using casacore::VarBufferedColumnarFile;
+using casacore::SimpleColumnarFile;
+
+template<typename ColumnarFile>
+void TestEmptyConstructor() {
+  ColumnarFile file;
+  BOOST_CHECK_EQUAL(file.NRows(), 0);
+  BOOST_CHECK_EQUAL(file.Filename(), "");
+  BOOST_CHECK(!file.IsOpen());
+
+  file = ColumnarFile();
+  BOOST_CHECK_EQUAL(file.NRows(), 0);
+  BOOST_CHECK_EQUAL(file.Filename(), "");
+  BOOST_CHECK(!file.IsOpen());
+}
+
+template<typename ColumnarFile>
+void TestCreateAndOpen() {
+  constexpr size_t kStrideA = 10;
+  const std::string filename = "columnar_file_test.tmp";
+  {
+    ColumnarFile file = ColumnarFile::CreateNew(filename, {}, kStrideA);
+    BOOST_CHECK_EQUAL(file.Stride(), kStrideA);
+    BOOST_CHECK_EQUAL(file.NRows(), 0);
+    BOOST_CHECK_EQUAL(file.Filename(), filename);
+    BOOST_CHECK(file.IsOpen());
+  }
+
+  // See if re-creating works
+  constexpr size_t kStrideB = 20;
+  ColumnarFile file = ColumnarFile::CreateNew(filename, {}, kStrideB);
+  BOOST_CHECK_EQUAL(file.Stride(), kStrideB);
+  BOOST_CHECK_EQUAL(file.NRows(), 0);
+  BOOST_CHECK_EQUAL(file.Filename(), filename);
+  BOOST_CHECK(file.IsOpen());
+  file.AddRows(43);
+  BOOST_CHECK_EQUAL(file.NRows(), 43);
+  file.DeleteRow();
+  BOOST_CHECK_EQUAL(file.NRows(), 42);
+  BOOST_CHECK(file.IsOpen());
+  // Close by assigning to empty
+  file = ColumnarFile();
+  BOOST_CHECK(!file.IsOpen());
```

Comment:
Here, you're just checking that calling `IsOpen()` after using default constructor yields `false`, which is already part of TestEmptyConstructor.

If you want to check that the file is really closed, you'd have to get the number of open file descriptors from the system before and after the assignment, for example. It doesn't seem trivial, though. Maybe you can try if `open(filename, ...)` with the proper options succeeds instead.

Reply:
Hmm really checking the nr of open files sounds a bit fragile. I think if the file is not closed properly, some of the other tests will fail, so I think that should be enough as a test.